Pacman_Search/search.py

```diff
@@ -87,7 +89,28 @@ def depthFirstSearch(problem):
     print "Start's successors:", problem.getSuccessors(problem.getStartState())
     """
     "*** YOUR CODE HERE ***"
-    util.raiseNotDefined()
+
+    st = util.Stack()
+    st.push([(problem.getStartState(), Directions.NORTH, 0)])
+
+    visitedNodes = []
+    actionList = []
+
+    while not st.isEmpty():
+        v = st.pop()
+        visitedNodes.append(v)
+
+        if problem.isGoalState(v):
+            return actionList
+
+        for action in problem.getSuccessors(v):
```

Comment:
action is a tuple and visitedNodes is a list of tuples. action[0] is actually the state, so checking for equality between a tuple and a state will fail at runtime